src/buildblock/ML_norm.cxx

```diff
@@ -768,25 +768,28 @@ void make_fan_data(FanProjData& fan_data,
 
   for (bin.segment_num() = proj_data.get_min_segment_num(); bin.segment_num() <= proj_data.get_max_segment_num();  ++ bin.segment_num())
   {
-    segment_ptr.reset(new SegmentBySinogram<float>(proj_data.get_segment_by_sinogram(bin.segment_num())));
-    
-    for (bin.axial_pos_num() = proj_data.get_min_axial_pos_num(bin.segment_num());
-	 bin.axial_pos_num() <= proj_data.get_max_axial_pos_num(bin.segment_num());
-	 ++bin.axial_pos_num())
-       for (bin.view_num() = 0; bin.view_num() < num_detectors_per_ring/2; bin.view_num()++)
-          for (bin.tangential_pos_num() = -half_fan_size;
-	       bin.tangential_pos_num() <= half_fan_size;
-               ++bin.tangential_pos_num())
-          {
-            int ra = 0, a = 0;
-            int rb = 0, b = 0;
-            
-            proj_data_info_ptr->get_det_pair_for_bin(a, ra, b, rb, bin);
-            
-            fan_data(ra, a, rb, b) =
-	      fan_data(rb, b, ra, a) =
-              (*segment_ptr)[bin.axial_pos_num()][bin.view_num()][bin.tangential_pos_num()];
+	  for (bin.timing_pos_num() = proj_data.get_min_tof_pos_num(); bin.timing_pos_num() <= proj_data.get_max_tof_pos_num(); ++ bin.timing_pos_num())
+	  {
+		segment_ptr.reset(new SegmentBySinogram<float>(proj_data.get_segment_by_sinogram(bin.segment_num(),bin.timing_pos_num())));
```

Comment:
this seems incorrect. fan-sums etc should sum over TOF bins. with current code. the fan data seems to be the one from the last TOF bin.

adaptations in ML_norm seems to need more thought. Can we just error out for now if its TOF?

Reply:
Sure. Done.